src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -501,6 +501,23 @@ public static function getMorphedModel($alias)
         return static::$morphMap[$alias] ?? null;
     }
 
+    /**
+     * Get the alias associated with a custom polymorphic class.
+     *
+     * @param  string  $className
+     * @return int|string|null
+     */
+    public static function getMorphAlias(string $className)
+    {
+        foreach (static::$morphMap as $alias => $morphedClass) {
+            if ($morphedClass === $className) {
+                return $alias;
+            }
+        }
+
+        return null;
```

Comment:
Would it not be easier to do the following?

```suggestion
        return array_flip(static::$morphMap)[$className] ?? null;
```

Reply:
Would it not be easier to use array_search?